src/main/kotlin/com/workos/users/users.kt

```diff
@@ -463,4 +464,73 @@ class UsersApi(private val workos: WorkOS) {
 
     return workos.post("/users/password_reset_challenge", CreatePasswordResetChallengeResponse::class.java, config)
   }
+
+  /**
+   * Parameters for the [authenticateUserWithPassword] method.
+   */
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  data class AuthenticateUserWithPasswordOptions @JvmOverloads constructor(
+    @JsonProperty("email") val email: String,
+    @JsonProperty("password") val password: String,
+    @JsonProperty("ip_address") val ipAddress: String? = null,
+    @JsonProperty("user_agent") val userAgent: String? = null,
+    @JsonProperty("start_session") val startSession: Boolean? = null,
+    @JsonProperty("expires_in") val expiresIn: Int? = null,
+    @JsonProperty("client_secret") val clientSecret: String? = null,
+    @JsonProperty("grant_type") val grantType: String = "password",
+  ) {
+    init {
+      require(email.isNotBlank()) { "Email is required." }
+      require(password.isNotBlank()) { "Password is required." }
+    }
+
+    /**
+     * Builder class for [AuthenticateUserWithPasswordOptions].
+     */
+    class AuthenticateUserWithPasswordOptionsBuilder {
+      private lateinit var email: String
+      private lateinit var password: String
+      private var ipAddress: String? = null
+      private var userAgent: String? = null
+      private var startSession: Boolean? = null
+      private var expiresIn: Int? = null
+      private var clientSecret: String? = null
+
+
+      fun email(value: String) = apply { this.email = value }
+      fun password(value: String) = apply { this.password = value }
+      fun ipAddress(value: String) = apply { this.ipAddress = value }
+      fun userAgent(value: String) = apply { this.userAgent = value }
+      fun startSession(value: Boolean) = apply { this.startSession = value }
+      fun expiresIn(value: Int) = apply { this.expiresIn = value }
+
+      fun clientSecret(value: String) = apply {this.clientSecret = value }
```

Comment:
We might be able to omit the setting function for this since we don't expect developers to set it this way.

Reply:
even though I'm technically setting it like a developer would?

Reply:
We actually aren't using this method currently. Rather it's being set with the `copy` method below.

Reply:
ah, that was my fault for not fully understanding how copy works. Neat.